plugins/woocommerce/src/Blocks/Domain/Services/Notices.php

```diff
@@ -42,11 +42,11 @@ public function __construct( Package $package ) {
 	 */
 	public function init() {
 		if ( wp_is_block_theme() ) {
-			add_filter( 'woocommerce_kses_notice_allowed_tags', [ $this, 'add_kses_notice_allowed_tags' ] );
 			add_filter( 'wc_get_template', [ $this, 'get_notices_template' ], 10, 5 );
-			add_action( 'wp_head', [ $this, 'enqueue_notice_styles' ] );
 		}
 
+		add_filter( 'woocommerce_kses_notice_allowed_tags', [ $this, 'add_kses_notice_allowed_tags' ] );
+		add_action( 'wp_head', [ $this, 'enqueue_notice_styles' ] );
```

Comment:
Thanks for mentioning this, @opr. Enqueueing the block notice styles for classic themes is necessary in case a merchant, that is using a classic theme, is using block notices. Without enqueueing the block notice styles, the notice would look like this:

<img width="840" alt="Screenshot 2024-03-05 at 15 32 36" src="https://github.com/woocommerce/woocommerce/assets/3323310/8a338d61-442a-4e3a-94df-a6487885b472">

The block notice styles, that we're enqueueing, add 2.7 kB to the page load. This impact should not be noticeable to merchants or sellers. 

<img width="2560" alt="Screenshot 2024-03-05 at 15 36 04" src="https://github.com/woocommerce/woocommerce/assets/3323310/1ec70819-425d-46bd-8249-19890b663286">

Reply:
> The block notice styles, that we're enqueueing, add 2.7 kB to the page load. This impact should not be noticeable to merchants or sellers.

Sure, but if we can avoid this somehow then it would be better. Can we think of a way of allowing classic themes to signal that they want to use block notices, a filter for example?

Before this PR, the notices were only enqueued wen `is_block_theme` is true, however imagine if we had something like

`apply_filters( 'woocommerce_blocks_use_block_notice_styles', is_block_theme() )`

What are your thoughts on that?